packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js

```diff
@@ -99,76 +165,151 @@ export function getDataInfo(data) {
+				selectedJsonpath={props.selectedJsonpath}
 			>
 				<NativeValue
 					data={data}
 					edit={isEdited}
 					jsonpath={jsonpath}
-					onClick={props.onClick}
+					onSelect={props.onSelect}
+					onEdit={props.onEdit}
 					onChange={props.onChange}
+					selectedJsonpath={props.selectedJsonpath}
 				/>
-			</LineItem>
+				{props.showType &&
+					<div className={`tc-object-viewer-linetype ${theme.lineType}`}>
+						({info.type})
+					</div>
+				}
+			</LineItem >
 		);
 	}
 
 	const iconName = isOpened ? 'talend-caret-down' : 'talend-chevron-left';
 	const iconTransform = isOpened ? null : 'rotate-180';
-	const btn = classNames(
-		'btn btn-xs btn-link',
-		theme.btn,
-	);
+	const decoratedLength = (info.type === 'array') ? `[${info.length}]` : `(${info.length})`;
+	const badgeClasses = classNames({
+		badge: true,
+		[theme.selectedBadge]: props.selectedJsonpath === jsonpath,
+	});
 
 	return (
-		<LineItem name={name} mouseOverData={{ data, isOpened, isEdited }}>
-			<span className={theme.hierarchical}>
-				<button
-					type="button"
-					className={btn}
-					onClick={e => props.onClick(e, { data, isOpened, jsonpath })}
-				>
-					<Icon name={iconName} transform={iconTransform} />
-					{info.type}
-					<sup className="badge">{info.length}</sup>
-				</button>
-				<ul className={!isOpened ? 'hidden' : null}>
-					{info.keys.map((key, i) => {
-						const jpath = getJSONPath(key, jsonpath, info.type);
-						return (
-							<li key={i}>
-								<Item
-									{...props}
-									data={data[key]}
-									name={key}
-									jsonpath={jpath}
-									opened={opened}
-									edited={edited}
-								/>
-							</li>
-						);
-					})}
-				</ul>
-			</span>
-		</LineItem>
+		<div>
+			<Icon
+				name={iconName}
+				transform={iconTransform}
+				className={theme.widerIconSelection}
+				onClick={(e) => {
+					e.stopPropagation();
+					props.onToggle(e, { data, isOpened, jsonpath });
+				}}
+			/>
+			<LineItem
+				name={name}
+				mouseOverData={{ data, isOpened, isEdited }}
+				onSelect={props.onSelect}
+				onToggle={props.onToggle}
+				jsonpath={jsonpath}
+				selectedJsonpath={props.selectedJsonpath}
+			>
+				<span className={theme.hierarchical}>
+					{props.showType ? (<div
+						className={`tc-object-viewer-linetype ${theme.lineType} `}
+						onClick={e => props.onSelect(e, { data, isOpened, jsonpath })}
+					>
+						({info.type})
+					</div>
+					) : null}
+					<TooltipTrigger className="offset" label={getDataAbstract(data)} tooltipPlacement="right">
+						<sup className={badgeClasses}>
+							{decoratedLength}
+						</sup>
+					</TooltipTrigger>
+					<ul className={!isOpened ? 'hidden' : `${theme.verticalLine} `}>
+
+						{info.keys.map((key, i) => {
+							const jpath = getJSONPath(key, jsonpath, info.type);
+							return (
+								<li key={i}>
+									<Item
+										{...props}
+										data={data[key]}
+										name={key}
+										jsonpath={jpath}
```

Comment:
Could be `jsonpath={getJSONPath(key, jsonpath, info.type)}` to avoid jpath allocation.

Reply:
Easier debug purpose